### ISSUE TITLE
Correctly handle and document how TypeAliasyType (pep685) work in SQLAlchemy

### ISSUE THREAD

**Issue by CaselIT** [sqlalchemy/sqlalchemy#11955]
As the title says, currently what there is not consistent and was organically developed without a clear api.

The main idea here is to thread type alias type as new type, since this is what python seems to suggest (eq of tho identical TypeAliasType is false)

This means that to use of them inside `Mapping` they need to be explicitly added to the type map. SQLAlchemy will not use the value the type alias refers to when checking the type.

This breaks what was added in #11305 but that's the only consistent way of doing it

**Comment by sqla-tester** [comment]
**Frazer McLean** has proposed a fix for this issue in the **main** branch:

**General improvement on annotated declarative** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5513

**Comment by sqla-tester** [comment]
**Federico Caselli** referenced this issue:

**General improvement on annotated declarative** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5602

**Comment by sqla-tester** [comment]
**Federico Caselli** has proposed a fix for this issue in the **main** branch:

**Remove fallbacks from the previous typing change** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5603

**Comment by zzzeek** [comment]
Trying to understand the change here in terms of that we are proposing breaking #11305 in the 2.0 series

using their test, this is what I have:

```py

from typing import Literal

from typing_extensions import TypeAlias
from typing_extensions import TypeAliasType

from sqlalchemy.orm import DeclarativeBase
from sqlalchemy.orm import Mapped
from sqlalchemy.orm import mapped_column
from sqlalchemy.schema import CreateTable

# deprecation warning
type Options484_A = Literal["one", "two"]

# no deprecation warning
Options484_B: TypeAlias = Literal["one", "two"]

# deprecation warning
Options484_C = TypeAliasType("Options484_C", Literal["one", "two"])


class Base(DeclarativeBase): ...

class MyTable(Base):
    __tablename__ = "my_table"
    id: Mapped[int] = mapped_column(primary_key=True)
    optiona: Mapped[Options484_A]
    optionb: Mapped[Options484_B]
    optionc: Mapped[Options484_C]

print(CreateTable(MyTable.__table__))
```


is that correct?   Why is `type Options484_A` so different than `Options484_B: TypeAlias`?  Are those different constructs?   When would I use one or the other ?

**Comment by CaselIT** [comment]
> Why is type Options484_A so different than Options484_B: TypeAlias? Are those different constructs? When would I use one or the other ?

Yes, Options484_B is a normal literal, it's not a TypeAliasType. That's exactly the same as doing `Options484_B = Literal["one", "two"]`

**Comment by CaselIT** [comment]
also I'm going with that python is giving us. TypeAliasType behaves the same as NewType, so we have to threat it as such in my opinion.

```py
from typing import Literal, TypeAlias

type t1 = Literal["one", "two"]
type t2 = Literal["one", "two"]

t3 = Literal["one", "two"]
t4 = Literal["one", "two"]
t5: TypeAlias = Literal["one", "two"]

print('t1==t2', t1==t2)
print('t1==t3', t1==t3)
print('t1==t5', t1==t5)
print('t3==t4', t3==t4)
print('t3==t5', t3==t5)
```
prints
```
t1==t2 False
t1==t3 False
t1==t5 False
t3==t4 True
t3==t5 True
```

**Comment by zzzeek** [comment]
oh right, the EQ test, right.    so silly

**Comment by CaselIT** [comment]
not sure why the coded it like that, but they must have had their reason

**Comment by sqla-tester** [comment]
**Mike Bayer** referenced this issue:

**edits and reorganization for union/pep695 typing docs** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5641

**Comment by sqla-tester** [comment]
**Mike Bayer** referenced this issue:

**edits and reorganization for union/pep695 typing docs** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5642
